UnitsNet/UnitConverter.cs

```diff
@@ -45,6 +45,14 @@ public static class UnitConverter
         private static readonly string UnitTypeNamespace = typeof(LengthUnit).Namespace;
         private static readonly Assembly UnitsNetAssembly = typeof(Length).GetAssembly();
 
+        private static readonly List<Type> QuantityTypes = UnitsNetAssembly.GetTypes()
```

Comment:
Since we are targeting .NET 4.0 we probably can't use `IReadOnlyList<T>` as I originally advised, I believe that came in .NET 4.5. I would recommend using an array and `.ToArray()` then, just to make it more clear that this is not supposed to be manipulated.